### PR TITLE
Navmazing conversion for PXEServer

### PR DESCRIPTION
**Pull request by psav**
Purpose or Intent
=================

navmazing conversion of PXEServer
{{pytest: cfme/tests/infrastructure/test_pxe.py cfme/tests/infrastructure/test_system_image_type.py cfme/tests/infrastructure/test_iso_datastore.py cfme/tests/infrastructure/test_customization_template.py}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 8bde876ef5fdbac4976d9f7c6b3a0316b251b50f:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9a21d3e00d814033f13947c9afce4fb2b146fd02:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a6206d890fa3e387a91162849328888ff9d12f63:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ca3da9645213a32c4cc8f0f107680e2307deaf2d:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit adf6752f1343a556681ab1246e021b866b6cb2ba:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 118132ae3bf412275b900cedfb28599c6f31186b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7177fc123379695e1921ebd56a449dba271a2bad:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2fbcaa687774045053c3d6c433793bc98c484975:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/infrastructure/pxe.py [hunk 1]
```diff
@@ -463,15 +463,23 @@ class SystemImageType(Updateable, Pretty):
...
+    HOST_OR_NODE = "Host / Node"
 
     def __init__(self, name=None, provision_type=None, appliance=None):
         self.appliance = appliance or get_or_create_current_appliance()
         self.name = name
         self.provision_type = provision_type
 
     def _form_mapping(self, create=None, **kwargs):
+        if kwargs.get('provision_type') == self.VM_OR_INSTANCE:
+            provision_type = "VM and Instance"
```
Comment (mfalesni) [inline cfme/infrastructure/pxe.py]: Why the check here? What if you want to write a negative test ...

Comment (psav) [inline cfme/infrastructure/pxe.py]: You're saying I could have just "used" the string from the const right?

Comment (mfalesni) [inline cfme/infrastructure/pxe.py]: Yes, whatever gets passed shall get to the UI, we can't make this the API to the product, protecting from the errors :smile:

Comment (mfalesni) [inline cfme/infrastructure/pxe.py]: So The constants are defined and users CAN use them to get whatever they need

**Comment by dajoRH** [comment]
Lint report for commit 17040f05a345de10163e11da341f99c1a4ad2968:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/infrastructure/pxe.py (hunk 1 above)
Comment (psav) [inline cfme/infrastructure/pxe.py]: Why can't we make this the API? If I Add an option for selecting <None> would that be better?

Comment (mfalesni) [inline cfme/infrastructure/pxe.py]: I don't think we should check strings that go in. How then can we do negative testing ...

Comment (psav) [inline cfme/infrastructure/pxe.py]: OK

Comment (psav) [inline cfme/infrastructure/pxe.py]: reverting that part

**Comment by dajoRH** [comment]
Lint report for commit 7cc70ed7a468a2ca42a24f0ce8f309d247dfc0db:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Pull request merged**